lib/Scanner/ICAPClient.php

```diff
@@ -15,28 +15,29 @@ public function __construct(string $host, int $port) {
 		$this->port = $port;
 	}
 
-	private function connect(): bool {
-		$this->socket = \socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
-
-		if (!\socket_connect($this->socket, $this->host, $this->port)) {
-			return false;
-		}
-
-		# McAfee seems to not properly close the socket once all response bytes are sent to the client
-		# we use a 5 sec time out on receiving data
-		if (!\socket_set_option($this->socket, SOL_SOCKET, SO_RCVTIMEO, ["sec" => 5, "usec" => 0])) {
-			return false;
+	private function connect(): void {
+		$this->writeHandle = @\stream_socket_client(
+			"tcp://{$this->host}:{$this->port}",
+			$errorCode,
+			$errorMessage,
+			5
+		);
+
+		if (!$this->writeHandle) {
+			throw new InitException(
+				\sprintf(
```

Comment:
String interpolation is preferred

Reply:
Ok... but it will be jerky due to a mixture of single/double quotes. I need double quotes inside the message itself so
`sprintf` looks more clean.

Reply:
you could remove the double quotes, or change them to single quotes. Taking into account that the neither the host or port will have a space, I doubt people will misinterpret the tcp string as multiple strings.

Reply:
nope... it's double quotes are expected here. E.g. https://github.com/owncloud/files_antivirus/blob/master/lib/Scanner/Socket.php#L60

Reply:
I had used a clean way to concat args in this repo years before it was decided that "the interpolation is preferred" :bearded_person:

Reply:
By the way the complete phrase sounds like "Interpolation is preferred to concatenation". And `sprintf` is NOT a concatenation - it's still interpolation. Anyway, you wanted it dirty - you got it dirty ;) 
Personally I dislike my last commit.